app/multitenant/common.go

```diff
@@ -13,10 +13,19 @@ func errorCode(err error) string {
 	return "500"
 }
 
-func timeRequest(method string, metric *prometheus.SummaryVec, f func() error) error {
+// timeRequest runs 'f' and records how long it took in the given Prometheus
+// metric.
+//
+// toStatusCode is a function that translates errors returned by 'f' into
+// HTTP-like status codes. If 'nil', uses a default function that returns
+// "500" for any error and "200" otherwise.
+func timeRequest(method string, metric *prometheus.SummaryVec, toStatusCode func(error) string, f func() error) error {
+	if toStatusCode == nil {
+		toStatusCode = errorCode
+	}
```

Comment:
I'm not a big fan of passing nil as a meaningful default.

Perhaps a second function (`timeRequest` vs `timeRequestStatus`) the first with the original semantics, the second which takes a fn -> string as a parameter)